### ISSUE TITLE
Improve `Detector` behavior

### ISSUE THREAD

**Issue by basil96** [alsolera/VideoF2B#42]
The current implementation of `videof2b.core.detection.Detector` is simple and fast, but it is also naive.  It simply follows the largest moving blob.  When that blob is not the aircraft of interest (such as big trucks on a nearby highway or field insects near the lens) we get gaps in the desired track points. This is annoying during a figure. We can do better.

There are at least two approaches we could implement:

1. In future 3D tracking work, continuously calculate the velocity and acceleration of the aircraft in world units.  If they exceed certain realistic thresholds between the current frame and the previous one (or averaged over a few previous frames), ignore the detected blob and try the next largest blob. Continue this search until the largest possible blob is found nearby, and use that one for the current frame. This approach requires `Detector` to have some interaction with the figure tracker.

2. The first approach would only work when 3D tracking is enabled, which is not always the case. An alternative is to perform similar work within `Detector` in image space. This might trigger falsely due to lack of 3D information, but I think it's worth trying. I expect some pitfalls that would need to be dealt with, such as when the aircraft moves out of frame for a while and some other moving object is detected in its absence. Another issue may be when a blob with larger area than the aircraft is detected sufficiently close to the aircraft; we don't want to start following it. Perhaps one way to alleviate those issues is to track the "velocity" vector in image space, and expect its direction and magnitude to stay within reasonable thresholds.  After all, the aircraft never suddenly "jumps" or reverses direction..  The search region around the last detected point could be adaptive based on such logic.

For now, approach 2 may be worth some research to see if something can be done to minimize gaps.
It's probably best to implement #28 before starting this work so that we have a baseline of current `Detector` performance.
